pkg/apimachinery/utils/meta.go

```diff
@@ -504,7 +508,20 @@ func (m *grafanaMetaAccessor) GetSpec() (spec any, err error) {
 			err = fmt.Errorf("error reading spec")
 		}
 	}()
-	spec = m.r.FieldByName("Spec").Interface()
+
+	f := m.r.FieldByName("Spec")
+	if f.IsValid() {
+		spec = f.Interface()
+		return
+	}
+
+	// Unstructured
+	u, ok := m.raw.(*unstructured.Unstructured)
+	if ok {
+		spec = u.Object["spec"]
```

Comment:
if the object doesn't have a spec should we return an error or `nil`? Right now we're inconsistent, returning `nil` for Unstructured and an error for anything else.
```suggestion
		spec, ok = u.Object["spec"]
```